File: array_and_hashing/contains_duplicate.py
def contains_duplicate_set(nums: list[int]) -> bool:
    seen = set()
    for num in nums:
        if nums in seen:
            return True
        seen.add(num)
    return False


def contains_duplicate_sort(nums: list[int]) -> bool:
    nums = nums.sort()

    return any(nums[i] == nums[i - 1] for i in range(1, len(nums)))

File: array_and_hashing/ContainsDuplicate.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {

    public static boolean containsDuplicateSet(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        for (int num : nums) {
            if (seen.contains(num))
                return true;
            seen.add(num);
        }
        return false;
    }

    public static boolean containsDuplicateSort(int[] nums) {
        int[] sorted = nums.clone();
        Arrays.sort(sorted);
        for (int i = 1; i < sorted.length; i++) {
            if (sorted[i] == sorted[i - 1])
                return true;
        }
        return false;
    }
}
